js/msp.js

```diff
@@ -965,8 +963,28 @@ var MSP = {
                 break;
 
             case MSP_codes.MSP_PID_ADVANCED:
-                if (semver.gte(CONFIG.apiVersion, "1.16.0")) {
-                    var offset = 0;
+                var offset = 0;
+                if (semver.gte(CONFIG.flightControllerVersion, "3.0.0")) {
```

Comment:
i agree, the two version types and the whole semver.gte... whatever is a total pain. we should really do feature checking, not absolute version checking (much like in the new iodef code) or even better, to generate both the fc serial_msp.c/h and msp.js code via some other xml or json or something, so they always match...

i picked flightControllerVersion because it's more transparent than the api version, since everybody talks about the fc firmware version, not the msp api verison, i think this makes way more sense.